source/blog/2018-07-19-ovn-dpdk.html.md

```diff
@@ -0,0 +1,90 @@
+
+While still experimental, OVN-DPDK in oVirt is now available in versions 4.2 and upwards.
+
+# What's changed?
+
+### Ansible DPDK host setup
+
+Host configuration for DPDK usage is now automated using Ansible. This primarly includes:
+* Hugepages configuration -- hugepage size and quantity in the kernel.
+* CPU partitioning.
+* Binding NICs to userspace drivers. 
+* OVS-DPDK related configuration (initialization, socket memory, pmd thread core binding, etc).
+
+The role is installed via Ansible galaxy:
+```
+# ansible-galaxy install oVirt.dpdk-setup
+```
+An example playbook:
+```
+- hosts: dpdk_host_0
+  vars:
+    pci_drivers:
+      "0000:02:00.1": "vfio-pci"
+      "0000:02:00.2": "igb"
+      "0000:02:00.3": ""
+    configure_kernel: true
+    bind_drivers: true
+    set_ovs_dpdk: false
+  roles:
+    - ovirt-ansible-dpdk-setup
+```
+The role is controlled by 3 boolean variables (all set to `true` by default) and a dictionary of devices and their drivers:
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning). WARNING: When set to `true` it is very likely to trigger reboot of the host, unless all required configuration is already pre-set.
+* `bind_drivers` -- determines whether the role should bind devices to their specified drivers.
+* `set_ovs_dpdk` -- determines whether the role should initialize and configure OVS for DPDK usage.
+* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. The empty string represents the default Linux driver for the specified device. Non-DPDK compatible drivers may be set; However, NICs with such drivers will not be configured for OVS-DPDK usage and will not have their CPU isolated in multiple NUMA environments.
+
+Additionally, there are several optional performance related variables:
+* `pmd_threads_count` -- determines the number of PMD threads per NIC (default: `1`).
+* `nr_2mb_hugepages` -- determines the number of 2MB hugepages, if 2MB hugepages are to be used (default: `1024`).
+* `nr_1gb_hugepages` -- determines the number of 1GB hugepages, if 1GB hugepages are to be used (default: `4`).
+* `use_1gb_hugepages` -- determines whether 1GB hugepages should be used, where 1GB hugepages are supported (default: `true`).
+
+For additional information, refer to the role's [repository](https://github.com/ovirt/ovirt-ansible-dpdk-setup/).
+
+### OVN Integration
+DPDK in oVirt now leverages the capabilities of OVN in the form of an OVN localnet. This allows for seamless connectivity across OVN networks, benefiting from OVN's software defined routers, switches, security groups, and ACL's.
+
+For more information about OVN localnet integration in oVirt, refer to oVirt's [Provider Physical Network RFE](https://ovirt.org/develop/release-management/features/network/provider-physical-network/).
+
+# Usage in oVirt
+1) Install `oVirt.dpdk-setup` Ansible role via ansible-galaxy:
```

Comment:
Using the engine vm console - I think it's more explicit

Reply:
You may install this wherever, it doesn't have to be Engine

Reply:
Ok I was unaware and do agree with you